dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -0,0 +1,358 @@
+                    {
+                        THROW_IF_FAILED(PushNotificationHelpers::GetComRegistrationFromRegistry(expectedAppServerArgs.data(), registeredClsid));
+                    }
+                    else
+                    {
+                        std::wstring storedComActivatorString{ RegisterComActivatorGuidAndAssets() };
+                        // Remove braces around the guid string
+                        registeredClsid = winrt::guid(storedComActivatorString.substr(1, storedComActivatorString.size() - 2));
+                    }
+
+                    auto notificationCallback{ winrt::create_instance<INotificationActivationCallback>(registeredClsid, CLSCTX_ALL) };
+                    THROW_IF_FAILED(notificationCallback->Activate(appUserModelId, invokedArgs, data, dataCount));
+                }
+                else
+                {
+                    m_notificationHandlers(Default(), activatedEventArgs);
+                }
+            }
+            else
+            {
+                auto appProperties = winrt::CoreApplication::Properties();
+                appProperties.Insert(ACTIVATED_EVENT_ARGS_KEY, activatedEventArgs);
+                SetEvent(GetWaitHandleForArgs().get());
+            }
+        }
+        else
+        {
+            m_notificationHandlers(Default(), activatedEventArgs);
+        }
+
+        return S_OK;
+    }
+    CATCH_RETURN()
+
+    void AppNotificationManager::Show(winrt::Microsoft::Windows::AppNotifications::AppNotification const& notification)
+    {
+        THROW_HR_IF(WPN_E_NOTIFICATION_POSTED, notification.Id() != 0);
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::INotificationProperties> notificationProperties = winrt::make_self<NotificationProperties>(notification);
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::INotificationTransientProperties> notificationTransientProperties = winrt::make_self<NotificationTransientProperties>(notification);
+
+        auto notificationAppId{ ::Microsoft::Windows::AppNotifications::Helpers::RetrieveNotificationAppId() };
+
+        DWORD notificationId = 0;
+        THROW_IF_FAILED(ToastNotifications_PostToast(notificationAppId.c_str(), notificationProperties.get(), notificationTransientProperties.get(), &notificationId));
+
+        THROW_HR_IF(E_UNEXPECTED, notificationId == 0);
+
+        winrt::Microsoft::Windows::AppNotifications::implementation::AppNotification* notificationImpl = get_self< winrt::Microsoft::Windows::AppNotifications::implementation::AppNotification>(notification);
+        notificationImpl->SetNotificationId(notificationId);
+    }
+
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressResult> AppNotificationManager::UpdateAsync(winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressData const data, hstring const tag, hstring const group)
```

Comment:
>UpdateAsync [](http://example.com/codeflow?start=148&length=11)

Must fix: This method does not "co_await resume_backgound" and thus executes synchronously with the caller. Maybe line 240 should have:

```
auto strong = get_strong();
co_await resume_background();
```